### DISCUSSION TITLE
UniversalAMDFormBrowser has not options for steam deck in last bios version.

### DISCUSSION THREAD

**Discussion by l33tm4st3r** [DavidS95/Smokeless_UMAF#19] [General]
Last update bios update (115) for steam deck, seems to disable or erase hiden options in bios. So UniversalAMDFormBrowser can't read it.

Any solution with a new UMAF or just we need to do a bios rollback ?

**Comment by Cederick** [comment]
Also just encounterd the issue. Would a BIOS dump help with figuring out what's going on?

**Comment by marcoand75** [comment]
Same problem with latest 115, downgraded back to 113, all work again, i hope someone fix it for 115

**Reply by bl4k1st** [reply]
> Same problem with latest 115, downgraded back to 113, all work again, i hope someone fix it for 115

could you please post a link to BIOS 113?

**Reply by marcoand75** [reply]
[F7A0113_sign.zip](https://github.com/DavidS95/Smokeless_UMAF/files/11130256/F7A0113_sign.zip)
https://gitlab.com/evlaV/jupiter-hw-support/-/commits/master/usr/share/jupiter_bios

**Reply by Cederick** [reply]
Isn't 115 necessary for the SMT fix?

**Reply by marcoand75** [reply]
> Isn't 115 necessary for the SMT fix?

I use steam deck tools for disable when need SMT in this moment. I don't know the change log of 115 bios version.
I prefer to wait to go ahead and try to find more information on 115 because with TDP unlock and increase in CPU and GPU frequency I'm so happy 😊

**Reply by syberphunk** [reply]
> Isn't 115 necessary for the SMT fix?

No. That's a linux kernel fix, not BIOS dependent.

**Comment by Cederick** [comment]
Well the big feature of 3.5 is fixing SMT so you don't have to disable SMT for the gain. You could just revert back to 3.4.6 then.

**Reply by marcoand75** [reply]
i forget to write it, i'm on win 11, i left steamOS after few days after i received the console.

**Comment by DavidS95** [comment]
See if the newest version of UMAF uploaded will solve the issue. If it still does not work I would recommend staying on an older bios.

**Reply by l33tm4st3r** [reply]
Not works for me. Thx anyway :)

**Reply by l33tm4st3r** [reply]
Also, RSEP.log

```
Welcome to SREP (Smokeless Runtime EFI Patcher) 0.1.4c
Opened SREP_Config

Config Size: 0x4E
Parsing Config

Stripping NewLine, Carriage and tab Return

Current Parsing Op LoadFromFS
OP Detected

Commnand LoadFromFS 
Current Parsing SuppressIFPatcher.efi
Found File SuppressIFPatcher.efi 
Current Parsing Op Exec
OP Detected

Commnand Exec 
Current Parsing Op LoadFromFS
OP Detected

Commnand LoadFromFS 
Current Parsing Loader.efi
Found File Loader.efi 
Current Parsing Op Exec
OP Detected

Commnand Exec 
Executing Load from FS

Loaded Image Success -> 6C778000
EXEC %r

Executing Load from FS

Loaded Image Success -> 6C777000
```

**Reply by Cederick** [reply]
Can report the same result

**Comment by BlankUserLing** [comment]
[115.txt](https://github.com/DavidS95/Smokeless_UMAF/files/11131996/113.txt)
[113.txt](https://github.com/DavidS95/Smokeless_UMAF/files/11131997/115.txt)
AMD CBS：
Offset has changed

**Reply by Cederick** [reply]
Hope that this is the nessesary information to create a fix.

**Reply by Haxorzz** [reply]
i wonder if anyone can create a pull request and fix it manually using this

**Comment by CryoByte33** [comment]
Has there been any progress on this? I'm hoping that we come up with a solution before it hits SteamOS stable 🙂

**Reply by Cederick** [reply]
So does reddit https://www.reddit.com/r/SteamDeck/comments/128nui6/steam_os_35_did_reset_bios_and_made_it_impossible

PS: Your CryoUtilities is unusuable when opened docked to a 1080p screen because it does not scale the window size correctly (waaay too large)

**Reply by CryoByte33** [reply]
Obviously, I was more asking if I can help 😅

As for CU, we're working on a fix, the issue should fix itself if you set the external monitor to primary, if not please open an issue!

**Comment by Mario-119** [comment]
Oh good I thought this was only me, hopefully this is fixable in due time

**Comment by ghost** [comment]
UPDATE:

The only addition to the steam Deck CBS/PBS Dxe in BIOS is:

```
  (*gBS->CreateEventEx)
                    (0x200,8,(EFI_EVENT_NOTIFY)&UnistallProtocollFunction,NotifyContext,
                     &EFI_EVENT_READY_TO_BOOT_GUID,&local_res10);
        }
```
Given that UMAF ofc is loaded after the READY_TO_BOOT, they used the READY_TO_BOOT event, to call a function that uninstall the form...

I could bypass this in varius way, by manually reinstalling the FORM, or by some other trick...
But for now I don't plan to...

I'm writing this here, just to make sure that no hope is created around an up-to-date UMAF..

**Reply by CryoByte33** [reply]
Given that a lot of users for this are for the Steam Deck (partially due to my advertising it 😅) is there any chance that you'd be open to add that function?

I can't pay too much, but I'd be willing to send some money your way if that'd help implement support for the Deck's new BIOS specifically.

**Reply by ghost** [reply]
We can try a simple way with the help of SREP, but I feel like that they added some other check on the READY_TO_BOOT  event..
that would make this fail, or make some setting not work anymore...

Even if it work I expect that will be patched in few revision
if U want I can bake a config for you to try... (if wanna to try out drop your discord handle, and will PM you)

A proper fix, require time and I don't want to waste too much on this one

**Reply by CryoByte33** [reply]
Sure, my Discord handle is CryoByte33#0345, I'm on EU time so I'm about to head to bed, but I'll be happy to work with you on it over the coming days!

We don't know if this was intentional by Valve or not, but they've given us no reason to think that it is. I'm of the opinion that they're probably changing the BIOS for new features in the future, and not because they want to lock people out.

**Reply by Cederick** [reply]
I'd like to add myself to the list of people that want to try this. Handle is Nexipal#0050 and I'm also off to bed soon.

**Reply by Mario-119** [reply]
I'd also like to be added to that list if possible, my handle on discord is Mario119#1247

**Reply by Haxorzz** [reply]
I'm also willing to test this patch but I don't have a discord

**Reply by Cederick** [reply]
Got the notification for an invite on Dicord but it was gone when I tried to accept it.

**Reply by Mario-119** [reply]
Same as Cederick, the invite disappeared for some reason before I could see it

**Reply by ghost** [reply]
https://gist.github.com/SmokelessCPUv2/8c1e6559031e199d9a678c9fe2ebf7d4

**Reply by RealDoomSlaya** [reply]
I'd like to be added to the list too, my handle is ⒹⓞⓞⓜⓈⓛⓐⓨⓐ▼▼▼#0485 

and @CryoByte33  I hope you don't mind; I've added you over on Discord to discuss a youtube problem I'm having. I've donated twice now, and the comment disappears or deletes itself for whatever reason. It includes a lot of OC'ing information that you might find handy and others. thanks!

**Reply by CryoByte33** [reply]
@RealDoomSlaya sorry about that, I accidentally denied it because I didn't see this first, try requesting me again 😅

**Reply by Cederick** [reply]
@SmokelessCPUv2 
The code does not run
sys/io.h is not present on Steam Deck

```c#
SD_Unlocker.c:1:10: fatal error: sys/io.h: No such file or directory
    1 | #include <sys/io.h>
      |          ^~~~~~~~~~
compilation terminated.

```

**Reply by rnwgnr** [reply]
This is awesome! Thank you very much.

I've compiled the binary on my main machine running ArchLinux and copied it over to the deck.
After executing the file and booting into BIOS both "AMD CBS" and "AMD PBS" options are there. A quick check on overclocking showed that it works as expected.

edit:
The change is persistent. The new options will be there even after multiple reboots.

**Reply by ghost** [reply]
@rnwgnr the change go away with a CMOS reset...
or just edit the code to have  outb(0x00,0x73); as last line instead of  outb(0x77,0x73);

the io port 72/73 is whats called extended CMOS region

**Reply by ghost** [reply]
> ```cs
> sys/io.h
> ```
@Cederick 
most likely you have some missing header/dependency...
that is standard linux header, so should somewhere in some missing package

**Reply by l33tm4st3r** [reply]
> https://gist.github.com/SmokelessCPUv2/8c1e6559031e199d9a678c9fe2ebf7d4

@SmokelessCPUv2 Thanks dude. Works like a charm.

**Reply by JDGBOLT** [reply]
Only thing that I'm going to say is that Valve is most likely going to patch this out in the near future. This will be even more digestible to ordinary folks considering it's run this binary and you unlock the uefi, no need for even flashing a utility on a usb stick, just run binary and go into uefi.

I would request that everyone don't just go crazy making tons of youtube videos and guides on undervolting the deck using this as there can be a very real possibility of the uninitiated to break things by just seeing some youtube video going "Free Performance!" with just running the binary, messing with settings purely based off of what is going on in the video, and then rebooting their deck to have it be a brick unless you reflash the uefi chip on the board.

And you will have people who reach this point without accepting the very real risks by just following along in a guide inevitably reach out to Valve who has been good about RMA'ing decks, perhaps a bit too good on this front, so people view it as just an out in case they mess up, which it really shouldn't be. This then makes them try to lock things down in order to protect people from themselves and not have to deal with the headache of people crying to them when they break things.

Honestly within the binary itself there should be warnings and other messages that you use this at your own risk, don't go to valve if you mess up, and here be dragons essentially.

**Reply by ghost** [reply]
Also i would add to this,
@rnwgnr (or @DavidS95 if you have the permission )can you remove the binary please, i think that if someone is not able to follow the instrction on how compile it, shouldn't use it..
Or at least let's don't have binary so public...

Also i would this now is far off topic and not related to umaf anymore, so @DavidS95 i think you can close it.

**Reply by TarnishedWorker** [reply]
Thank you for the solution, and it works!

I noticed that for the settings and values that have already been presented in the factory BIOS menu, if I changed those in the CBS menu on my own, they will be overridden back by the factory settings.

For example, Steam Deck's factory BIOS allows us to change the TDP settings under the Advanced menu, up to maximum input of 15,000mw. Now if I go into the CBS menu and update the TDP settings (slow and fast) to go beyond 15,000mv, and when I hit "save & exit", it will actually save the Steam Deck's default maximum values (15,000mv) rather than the values I prefer (e.g., 25,000mv).

Is there any way we can solve this problem?

**Reply by rnwgnr** [reply]
> Also i would add to this, @rnwgnr (or @DavidS95 if you have the permission )can you remove the binary please, i think that if someone is not able to follow the instrction on how compile it, shouldn't use it.. Or at least let's don't have binary so public...

I have removed the binary.

thanks for your efforts anyway 👍
